lib/dfdutils/printdfd.c

```diff
@@ -31,25 +684,47 @@ void printDFD(uint32_t *DFD)
     int samples = (KHR_DFDVAL(BDB, DESCRIPTORBLOCKSIZE) - 4 * KHR_DF_WORD_SAMPLESTART) / (4 * KHR_DF_WORD_SAMPLEWORDS);
     int sample;
     int model = KHR_DFDVAL(BDB, MODEL);
+
+#define PRINT_ENUM(VALUE, TO_STRING_FN) {                        \
+        int value = VALUE;                                       \
+        const char* str = TO_STRING_FN(value);                   \
+        if (str)                                                 \
+            printf("%s", str);                                   \
+        else                                                     \
+            printf("%d", value);                                 \
+    }
+
     printf("DFD total bytes: %d\n", DFD[0]);
-    printf("BDB descriptor type 0x%04x vendor id = 0x%05x\n",
-           KHR_DFDVAL(BDB, DESCRIPTORTYPE),
-           KHR_DFDVAL(BDB, VENDORID));
-    printf("Descriptor block size %d (%d samples) versionNumber = 0x%04x\n",
+
+    printf("BDB descriptor type: ");
+    PRINT_ENUM(KHR_DFDVAL(BDB, DESCRIPTORTYPE), dfdToStringDescriptorType);
```

Comment:
This comment applies to both text and JSON output.

We have interest in KTX v2 from the USD folks. If they adopt they will almost certainly want to add ICC profile support which will be in the form of an extended DFD. Therefore this program needs to be prepared for multiple extension blocks within the descriptor (using `totalDfdSize` to determine if there are extras) and, for extension blocks, to print the values of the `KHR_DF_KHR_DESCRIPTORTYPE_NEEDED_FOR_WRITE_BIT` and `KHR_DF_KHR_DESCRIPTORTYPE_NEEDED_FOR_DECODE_BIT` in the `descriptorType` field.

Reply:
I thought we agreed on the `descriptorType` to be treated as an enum, as it is one. Having to write out the indicator bits too means the output needs to be converted to a flag type (i.e. array in the JSON output), which is kind of weird.

Are you sure that's needed, considering that the appropriate `descriptorType` enum values that do contain those bits set too would already convey the same information anyway?

Reply:
> I thought we agreed on the `descriptorType` to be treated as an enum.

We did. Since then I've been learning more about extension blocks. I've sent a message to the Data Format spec author to ask about the enum values.

Reply:
These bits were retrofitted to the spec. The DF author is playing semantic games between a spec. clarification (the bits are part of the descriptor Id) and a spec. change (the descriptor Id becomes 13 bits instead of 15). The usage of these and interaction with registration of vendor extension numbers needs to be, er, clarified. We can come back to this later.